pydantic/env_settings.py

```diff
@@ -44,66 +46,110 @@ def _build_values(
+        case_sensitive = False
 
-        if not secrets_path.exists():
-            raise SettingsError(f'directory "{secrets_path}" does not exist')
-        if not secrets_path.is_dir():
-            raise SettingsError(f'secrets_dir must reference a directory, not a {path_type(secrets_path)}')
+        @classmethod
+        def prepare_field(cls, field: ModelField) -> None:
+            env_names: Union[List[str], AbstractSet[str]]
+            field_info_from_config = cls.get_field_info(field.name)
 
-        for field in self.__fields__.values():
-            for env_name in field.field_info.extra['env_names']:
-                path = secrets_path / env_name
-                if path.is_file():
-                    secrets[field.alias] = path.read_text().strip()
-                elif path.exists():
+            env = field_info_from_config.get('env') or field.field_info.extra.get('env')
+            if env is None:
+                if field.has_alias:
                     warnings.warn(
-                        f'attempted to load secret file "{path}" but found a {path_type(path)} instead.',
-                        stacklevel=4,
+                        'aliases are no longer used by BaseSettings to define which environment variables to read. '
+                        'Instead use the "env" field setting. '
+                        'See https://pydantic-docs.helpmanual.io/usage/settings/#environment-variable-names',
+                        FutureWarning,
                     )
+                env_names = {cls.env_prefix + field.name}
+            elif isinstance(env, str):
+                env_names = {env}
+            elif isinstance(env, (set, frozenset)):
+                env_names = env
+            elif sequence_like(env):
+                env_names = list(env)
+            else:
+                raise TypeError(f'invalid field env: {env!r} ({display_as_type(env)}); should be string, list or set')
 
-        return secrets
+            if not cls.case_sensitive:
+                env_names = env_names.__class__(n.lower() for n in env_names)
+            field.field_info.extra['env_names'] = env_names
+
+        @classmethod
+        def customise_sources(
+            cls,
+            init_settings: SettingsSourceCallable,
+            env_settings: SettingsSourceCallable,
+            file_secret_settings: SettingsSourceCallable,
+        ) -> Tuple[SettingsSourceCallable, ...]:
+            return init_settings, env_settings, file_secret_settings
+
+    __config__: Config  # type: ignore
+
+
+class InitSettingsSource:
+    __slots__ = ('init_kwargs',)
+
+    def __init__(self, init_kwargs: Dict[str, Any]):
+        self.init_kwargs = init_kwargs
 
-    def _build_environ(
-        self, _env_file: Union[Path, str, None] = None, _env_file_encoding: Optional[str] = None
-    ) -> Dict[str, Optional[str]]:
+    def __call__(self, settings: BaseSettings) -> Dict[str, Any]:
+        return self.init_kwargs
```

Comment:
might be useful to implement repr here to help people implementing their own `customise_sources`

Just something like `f'InitSettingsSource(init_kwargs={self. init_kwargs!r})'`

Reply:
and for `EnvSettingsSource` and `SecretsSettingsSource`

Reply:
Good idea !